src/build.js

```diff
@@ -61,11 +61,17 @@ const followBuild = async ({ appSlug, buildSlug, client }, options = {}) => {
       lastActive = now;
     }
 
+    // Sometimes build might have empty log_chunks, have is_archived set to false, have non-empty timestamp, but be aborted
+    // (because of bitrise timeout for example). Don't follow such builds forever:
+    attributes = await describeBuild({appSlug, buildSlug, client});
+    if (attributes.status === 3 && !response.data.is_archived && response.data.log_chunks.length === 0) {
+      process.stdout.write('Build has been aborted, not polling logs any more\n');
+      throw new Error(`Build ${appSlug}/${buildSlug} aborted`);
+    }
+
     timestamp = response.data.timestamp;
   } while (timestamp);
 
-  const attributes = await describeBuild({ appSlug, buildSlug, client });
-
   if (attributes.status > 1) {
```

Comment:
Will this fail if [the `break` on line 50 is hit](https://github.com/lifeomic/bitrise/pull/33/files#diff-abc8f6a58535fbae6adcdf1f8348cb6bed037902afdf5731f0070dd0cdb28753R50) before fetching the attributes?  This file needs TypeScript for safety

Reply:
Great catch! I've fixed this. Turns out one of the unit tests was failing because of this ("following a successful build that has already finished prints the log output"), but somehow I didn't notice this. Would you mind if I convert this to typescript in a follow-up PR?

Reply:
I like the idea of a TypeScript conversion